sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -438,7 +438,7 @@ public virtual Response<TableItem> CreateIfNotExists(CancellationToken cancellat
                     cancellationToken);
                 return Response.FromValue(response.Value as TableItem, response.GetRawResponse());
             }
-            catch (RequestFailedException ex) when (ex.Status == (int)HttpStatusCode.Conflict)
+            catch (RequestFailedException ex) when (ex.Status == (int)HttpStatusCode.Conflict && ex.ErrorCode != TableErrorCode.TableBeingDeleted)
             {
                 return default;
```

Comment:
Wait, are we returning null from this method? It's against the guidelines...

Reply:
yes - This must have been missed before.